src/movepick.cpp

```diff
@@ -73,8 +73,8 @@ MovePicker::MovePicker(const Position& p, Move ttm, Depth d, const ButterflyHist
 
 /// MovePicker constructor for quiescence search
 MovePicker::MovePicker(const Position& p, Move ttm, Depth d, const ButterflyHistory* mh,
-                       const CapturePieceToHistory* cph, Square rs)
-           : pos(p), mainHistory(mh), captureHistory(cph), recaptureSquare(rs), depth(d) {
+                       const CapturePieceToHistory* cph, const PieceToHistory** ch, Square rs)
```

Comment:
I think you can remove the double indirection here (`const PieceToHistory**` -> `const PieceToHistory*`) and just pass `(ss-1)->contHistory` to the constructor below (only the first member of the array is used by `MovePicker`). Or am I missing something?

Reply:
Yes that would be possible. But in the Movepicker object only a pointer to the array of the counter move histories is currently used (const PieceToHistory** contHistory;). There itself is no array of the four cmh pointers defined.
We would have to define it there like PieceToHistory* contHistory[4]; and modify both constructors. That  seems not so elegant for me (but is debatable).

But an important reason is that we could easy test using the other histories (which i have already started)

Reply:
Oh, now I see that I missed that other constructors use the same pattern already and that in other places more elements of the array are used. You can disregard my previous comment then :)

Reply:
I have looked at this more and we can simply fake the array by using in the second Movepicker constructor  contHistory(&ch) in the initiliazer list (so only the first index exists but the others are not used). But i would rewrite this only after a period where we have tested out the possibilities of the other histories.

Thanks for the inspiration!